Update Hd44780.java

Cyber-One opened the pull request:
Lots of changes to make it more reliable.
Also added more functions and lots of javaDocs :-)
Also removed the delay value, this can be done in the user program.

Cyber-One committed:
Update Hd44780.java

Lots of changes to make it more reliable.
Also added more functions and lots of javaDocs :-)

Cyber-One committed:
Update Pcf8574.java

Small bug fix.
made int readRegister() a public method and added a javadocs for it.
This will be needed for the hd44780 service as it uses this method.

supertick commented on src/main/java/org/myrobotlab/service/Hd44780.java:
```diff
@@ -196,97 +193,288 @@ public void display(String string, int line) {
         break;
       case 3:
-        lcdWrite((byte) 0xD4);
+      setDDRAMaddress((byte) 0x3C);
```
The convention we follow with the myrobotlab is camel back case including abbreviations
So setDDRAMaddress should be setDdramAddress 
just like setXML would be setXml

supertick commented on src/main/java/org/myrobotlab/service/Hd44780.java:
```diff
@@ -296,6 +484,10 @@ public void setBackLight(boolean status) {
+  // This is a dangerous method to the HD44780 as anything written here can destabilize it.
+  // not yet removed but should be depreciated or removed if not being used.
+  @Deprecated // use lcdWriteCmd(byte cmd) or lcdWriteData(byte data)
   public void writeRegister(byte cmd) {
```
its used internally so it shouldn't be deprecated unless its mis-used,
but you should change the public to private

Cyber-One committed:
Update Hd44780.java

Change the setDdramAddress function name as requested and removed un-used public function that gave direct access to the PCF write register.
